packages/migration-graph-shared/src/entities/package-graph.ts

```diff
@@ -93,6 +95,13 @@ export class PackageGraph {
 
       const sourcePath = resolveRelative(baseDir, m.source);
 
+      if (this.isFileExternalToPackage(sourcePath)) {
+        DEBUG_CALLBACK(`${sourcePath} is out of package, not adding to module graph.`);
+        // Should resolve path completely relativeto the project and find which package it belongs to.
+        // Ask project which package does this belong maybe create an edge?
+        return;
+      }
+
```

Comment:
This is the thing that resolve #439

Reply:
If we create an edge we potentially create a cycle since the root package.json is typically a parent to all found packages within a workspace.

Reply:
Should we warn in these situations?

Reply:
I can aggregate these issues or console.warn immediately.

Reply:
Yes we should warn, we don't have a strategy for yielding information to the user when we construct a graph. 

I think in another PR I can refactor the migration-graph packages to capture and store messages and populate that in the migration-strategy that is used in the CLI.

Alternatively a simple console.warn for now can be a temp solution.